src/integrationTest/java/org/opensearch/security/http/AsyncTests.java

```diff
@@ -0,0 +1,115 @@
+        String indexName = "test-index";
+        final String invalidateCachePath = "/_plugins/_security/api/cache";
+        final String nodesPath = "/_nodes";
+        final String bulkPath = "/_bulk";
+        final String document = ("{ \"index\": { \"_index\": \"" + indexName + "\" }}\n{ \"foo\": \"bar\" }\n").repeat(5);
+        final int parallelism = 5;
+        final int totalNumberOfRequests = 30;
+
+        try (final TestRestClient client = cluster.getRestClient()) {
+            IndexOperationsHelper.createIndex(cluster, indexName);
+
+            final CountDownLatch countDownLatch = new CountDownLatch(1);
+
+            final var cacheInvalidationRequests = AsyncActions.generate(() -> {
+                countDownLatch.await();
+                System.err.println("Generation triggered invalidate cache requests");
+                final HttpDelete delete = new HttpDelete(client.getHttpServerUri() + invalidateCachePath);
+                return client.executeRequest(delete, getBasicAuthHeader(ADMIN_USER.getName(), ADMIN_USER.getPassword()));
+            }, parallelism, totalNumberOfRequests);
+
+            final var bulkRequests = AsyncActions.generate(() -> {
+                countDownLatch.await();
+                System.err.println("Generation triggered bulk requests");
+                final HttpPost post = new HttpPost(client.getHttpServerUri() + bulkPath);
+                post.setEntity(new ByteArrayEntity(document.getBytes(StandardCharset.UTF_8), ContentType.APPLICATION_JSON));
+                return client.executeRequest(post, getBasicAuthHeader(ADMIN_USER.getName(), ADMIN_USER.getPassword()));
+            }, parallelism, totalNumberOfRequests);
+
+            final var nodesRequests = AsyncActions.generate(() -> {
+                countDownLatch.await();
+                System.err.println("Generation triggered node requests");
+                final HttpGet get = new HttpGet(client.getHttpServerUri() + nodesPath);
+                return client.executeRequest(get, getBasicAuthHeader(ADMIN_USER.getName(), ADMIN_USER.getPassword()));
+            }, parallelism, totalNumberOfRequests);
+
+            // Make sure all requests start at the same time
+            countDownLatch.countDown();
+
+            List<CompletableFuture<HttpResponse>> allRequests = new ArrayList<CompletableFuture<HttpResponse>>();
+            allRequests.addAll(cacheInvalidationRequests);
+            allRequests.addAll(bulkRequests);
+            allRequests.addAll(nodesRequests);
+            Collections.shuffle(allRequests);
+            AsyncActions.getAll(allRequests, 150, TimeUnit.SECONDS).forEach((response) -> {
+                if (response.getStatusCode() == HttpStatus.SC_INTERNAL_SERVER_ERROR) {
+                    System.out.println("Response body: " + response.getBody());
+                }
+                assertThat(response.getStatusCode(), equalTo(HttpStatus.SC_OK));
```

Comment:
Nit: The built in assert on response handles this case
```suggestion
                response.assertStatusCode(HttpStatus.SC_OK);
```

Reply:
Thanks! I need to get used to using that ;). Updated.